cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -137,45 +153,107 @@ public void configure(Map<String, ?> configs) {
-  throws CruiseControlMetricsReporterException {
+  protected NewTopic createNewTopicFromReporterConfig(CruiseControlMetricsReporterConfig reporterConfig)
+      throws CruiseControlMetricsReporterException {
     String cruiseControlMetricsTopic =
         reporterConfig.getString(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_CONFIG);
     Integer cruiseControlMetricsTopicNumPartition =
         reporterConfig.getInt(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_NUM_PARTITIONS_CONFIG);
     Short cruiseControlMetricsTopicReplicaFactor =
         reporterConfig.getShort(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_REPLICATION_FACTOR_CONFIG);
-
     if (cruiseControlMetricsTopicReplicaFactor <= 0 || cruiseControlMetricsTopicNumPartition <= 0) {
       throw new CruiseControlMetricsReporterException("The topic configuration must explicitly set the replication factor and the num partitions");
     }
+    NewTopic newTopic = new NewTopic(cruiseControlMetricsTopic, cruiseControlMetricsTopicNumPartition, cruiseControlMetricsTopicReplicaFactor);
 
-    return new NewTopic(cruiseControlMetricsTopic, cruiseControlMetricsTopicNumPartition, cruiseControlMetricsTopicReplicaFactor);
+    Map<String, String> config = new HashMap<>(2);
+    config.put(LogConfig.RetentionMsProp(),
+               Long.toString(reporterConfig.getLong(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_RETENTION_MS_CONFIG)));
+    config.put(LogConfig.CleanupPolicyProp(), CRUISE_CONTROL_METRICS_TOPIC_CLEAN_UP_POLICY);
+    newTopic.configs(config);
+    return newTopic;
   }
 
-  private void createCruiseControlMetricsTopic() {
+  protected void createCruiseControlMetricsTopic() {
     try {
       final CreateTopicsResult createTopicsResult = _adminClient.createTopics(Collections.singletonList(_newTopic));
       createTopicsResult.values().get(_newTopic.name()).get();
-      LOG.info("Cruise Control metrics topic created: {}", _cruiseControlMetricsTopic);
+      LOG.info("Cruise Control metrics topic {} is created.", _cruiseControlMetricsTopic);
     } catch (InterruptedException | ExecutionException e) {
-      if (!(e.getCause() instanceof TopicExistsException)) {
-        LOG.error("Unable to create Cruise Control topic", e);
+      LOG.warn("Unable to create Cruise Control metrics topic {}.", _cruiseControlMetricsTopic, e);
+    }
+  }
+
+  protected void maybeUpdateCruiseControlMetricsTopic() {
+    maybeUpdateTopicConfig();
+    maybeIncreaseTopicPartitionCount();
+  }
+
+  protected void maybeUpdateTopicConfig() {
+    try {
+      // Retrieve topic config to check and update.
+      ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, _cruiseControlMetricsTopic);
+      DescribeConfigsResult describeConfigsResult = _adminClient.describeConfigs(Collections.singleton(topicResource));
+      Config topicConfig = describeConfigsResult.values().get(topicResource).get();
```

Comment:
For `describeConfigsResult` and `alterConfigsResult`, we seem to be blocking indefinitely.
Can we add timeouts to relevant get calls to avoid potential deadlock?